server/api/cvrs.py

```diff
@@ -381,12 +394,11 @@ def process_cvr_file(
     jurisdiction = Jurisdiction.query.get(jurisdiction_id)
 
     def process() -> None:
-        header_lines = 4 if jurisdiction.cvr_file_type == CvrFileType.DOMINION else 1
-        total_lines = len(jurisdiction.cvr_file.contents.splitlines()) - header_lines
-        emit_progress(0, total_lines)
-
-        if jurisdiction.cvr_file.contents == "":
-            raise UserError("CVR file cannot be empty.")
+        # Ideally, the CVR should have the same number of ballots as the
+        # manifest, so we can use that as an approximation of the file parsing
+        # progress since we're streaming the file and don't know the size up front.
+        total_records = jurisdiction.manifest_num_ballots
```

Comment:
It seems likely that we could get the file size in bytes, but I'm not sure how we would get it in lines, which is the current progress denominator. It might be possible to figure out a way to make the stream also report its progress in bytes, but since it's a few layers deep in generators, it seems tricky.

Given that all this functionality does is show a progress bar to the user, an approximate level of accuracy should be sufficient IMO